examples/tutorials/nb_python/ECCV_2020_Interactivity.py

```diff
@@ -0,0 +1,2093 @@
+        if show_waypoint_indicators:
+            sim.set_translation(continuous_path_follower.waypoint, vis_ids[0])
+
+        previous_rigid_state = sim.get_rigid_state(locobot_id)
+
+        # set velocities based on relative waypoint position/direction
+        track_waypoint(
+            continuous_path_follower.waypoint,
+            previous_rigid_state,
+            vel_control,
+            dt=time_step,
+        )
+
+        # manually integrate the rigid state
+        target_rigid_state = vel_control.integrate_transform(
+            time_step, previous_rigid_state
+        )
+
+        # snap rigid state to navmesh and set state to object/agent
+        end_pos = sim.step_filter(
+            previous_rigid_state.translation, target_rigid_state.translation
+        )
+        sim.set_translation(end_pos, locobot_id)
+        sim.set_rotation(target_rigid_state.rotation, locobot_id)
+
+        # Check if a collision occured
+        dist_moved_before_filter = (
+            target_rigid_state.translation - previous_rigid_state.translation
+        ).dot()
+        dist_moved_after_filter = (end_pos - previous_rigid_state.translation).dot()
+
+        # NB: There are some cases where ||filter_end - end_pos|| > 0 when a
+        # collision _didn't_ happen. One such case is going up stairs.  Instead,
+        # we check to see if the the amount moved after the application of the filter
+        # is _less_ than the amount moved before the application of the filter
+        EPS = 1e-5
+        collided = (dist_moved_after_filter + EPS) < dist_moved_before_filter
+
+        gripper.sync_states()
+        # run any dynamics simulation
+        sim.step_physics(time_step)
+
+        # render observation
+        observations.append(sim.get_sensor_observations())
+
+# release
+gripper.release()
+start_time = sim.get_world_time()
+while sim.get_world_time() - start_time < 2.0:
+    sim.step_physics(time_step)
+    observations.append(sim.get_sensor_observations())
+
+# video rendering with embedded 1st person view
+video_prefix = "fetch"
+if make_video:
+    make_video_cv2(
+        observations,
+        prefix=video_prefix,
+        open_vid=show_vid,
+        multi_obs=True,
+        fps=1.0 / time_step,
+    )
+
+# remove locobot while leaving the agent node for later use
+sim.remove_object(locobot_id, delete_object_node=False)
+remove_all_objects(sim)
+
+# %% [markdown]
+# # Advanced Features
+# This section contains advanced examples/demos of Habitat-sim interactivity.
+
+# %%
+# @title Initialize Simulator and Load Scene { display-mode: "form" }
+sim_settings = make_default_settings()
+sim_settings["scene"] = "./data/scene_datasets/mp3d/17DRP5sb8fy/17DRP5sb8fy.glb"
+sim_settings["sensor_pitch"] = 0
+
+make_simulator_from_settings(sim_settings)
+
+# %% [markdown]
+# ## Advanced Topic : Asset and Object Customization
+#
+# TODO: a Colab form to edit all fields of an object template and register it.
```

Comment:
@aclegg3 Is this TODO still valid?

Reply:
No, i missed the todo.  unfortunately I am unable to commit directly to repo so I need a kind and gallant soul to nuke that for me :)